bindings/python/pymongoarrow/api.py

```diff
@@ -264,8 +270,15 @@ def _tabular_generator(tabular):
             for row in i.to_pylist():
                 yield row
     elif isinstance(tabular, DataFrame):
-        for i in tabular.to_dict("records"):
-            yield i
+        for row in tabular.to_dict("records"):
+            yield row
+    elif isinstance(tabular, dict):
+        iter_dict = {k: iter(v.tolist()) for k, v in tabular.items()}
```

Comment:
Does this put the whole array into memory?  Maybe look into https://numpy.org/doc/stable/reference/generated/numpy.nditer.html#numpy.nditer

Reply:
Good catch. Fixed.